smac/epm/partial_sparse_gaussian_process.py

```diff
@@ -0,0 +1,874 @@
+                 ):
+        """
+        Partial Sparse Gaussian process model. It is composed of two models: an Exact GP to descirbe the data
+        distribution inside a subregion and an Approximate GP to approxiamte the data distribution outside a
+        subregion.
+
+        The GP hyperparameterŝ are obtained by optimizing the marginal log likelihood and optimize with botorch
+
+        Parameters
+        ----------
+        types : List[int]
+            Specifies the number of categorical values of an input dimension where
+            the i-th entry corresponds to the i-th input dimension. Let's say we
+            have 2 dimension where the first dimension consists of 3 different
+            categorical choices and the second dimension is continuous than we
+            have to pass [3, 0]. Note that we count starting from 0.
+        bounds : List[Tuple[float, float]]
+            bounds of input dimensions: (lower, uppper) for continuous dims; (n_cat, np.nan) for categorical dims
+        seed : int
+            Model seed.
+        kernel : george kernel object
+            Specifies the kernel that is used for all Gaussian Process
+        num_inducing_points: int
+            Number of inducing points
+        likelihood: Optional[GaussianLikelihood]
+            Likelihood values
+        normalize_y : bool
+            Zero mean unit variance normalization of the output values, when the model is a partial sparse GP model,
+        n_opt_restart : int
+            Number of restarts for GP hyperparameter optimization
+        instance_features : np.ndarray (I, K)
+            Contains the K dimensional instance features of the I different instances
+        pca_components : float
+            Number of components to keep when using PCA to reduce dimensionality of instance features. Requires to
+            set n_feats (> pca_dims).
+        """
+        super(PartialSparseGaussianProcess, self).__init__(configspace=configspace,
+                                                           types=types,
+                                                           bounds=bounds,
+                                                           seed=seed,
+                                                           kernel=kernel,
+                                                           likelihood=likelihood,
+                                                           normalize_y=normalize_y,
+                                                           n_opt_restarts=n_opt_restarts,
+                                                           instance_features=instance_features,
+                                                           pca_components=pca_components,
+                                                           )
+        self.bounds_cont = bounds_cont,
+        self.bounds_cat = bounds_cat,
+        self.num_inducing_points = num_inducing_points
+
+    def update_attribute(self, **kwargs: typing.Any):
```

Comment:
The type hint is wrong. kwargs (i.e. *k*ey-*w*ord *arg*uments) will always be a dictionary.
The entries however can be of type `typing.Any`